tree/dataframe/test/dataframe_snapshot.cxx

```diff
@@ -1455,4 +1455,67 @@ TEST(RDFSnapshotMore, ZeroOutputEntriesMT)
    gSystem->Unlink(fname);
 }
 
+// Test for custom basket size in Snapshot, ROOT - #17418
+TEST(RDFSnapshotMore, CustomBasketSize)
+{
+    // Create a simple ROOT file with a TTree
+    auto input_file = TFile::Open("input_file.root", "RECREATE");
+    auto tree = new TTree("tree", "Test Tree");
+
+    float value = 0;
+    tree->Branch("branch_x", &value);
+
+    // Fill the tree with some data
+    for (int i = 0; i < 1000; ++i) {
+        value = i;
+        tree->Fill();
+    }
+
+    input_file->Write();
+    input_file->Close();
+
+    // Read the file back using RDataFrame
+    ROOT::RDataFrame df("tree", "input_file.root");
+
+    // Define a new column
+    auto df_with_new_column = df.Define("branch_x_new", [](float x) { return x; }, {"branch_x"});
+
+    // Create RSnapshotOptions with a custom basket size
+    ROOT::RDF::RSnapshotOptions options;
+    options.fBasketSize = 2048; // Set custom basket size (2KB)
+
+    // Snapshot with custom basket size
+    std::string output_file_custom = "output_file_custom_basket.root";
+    df_with_new_column.Snapshot("tree", output_file_custom, {"branch_x", "branch_x_new"}, options);
```

Comment:
This will for sure not work as you expect. All the branches involved in this computation graph are of type `float`. You are not modifying anything in `ActionHelpers.hxx` to act upon branches of this type. This is the other function template I was referring to. Currently you are just passing the `basketSize` as parameter, you need to actually make use of it in the function.